gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/integrated/PartitionStrategy.feature

```diff
@@ -0,0 +1,458 @@
+    And the traversal of
+      """
+      g.withStrategies(new PartitionStrategy(partitionKey: "_partition", writePartition: "a", readPartitions: ["a"])).
+        inject(0).
+        mergeV(xx1)
+      """
+    When iterated to list
+    Then the result should have a count of 1
+    And the graph should return 1 for count of "g.V().has(\"person\",\"name\",\"alice\").has(\"_partition\",\"a\")"
+    And the graph should return 1 for count of "g.V()"
+
+  @WithPartitionStrategy
+  Scenario: g_withStrategiesXPartitionStrategyXwrite_a_read_aXX_mergeE
+    Given the empty graph
+    And the graph initializer of
+      """
+      g.addV("person").property("_partition","a").property("name","alice").
+        addV("person").property("_partition","a").property("name","bob")
+      """
+    And using the parameter xx1 defined as "m[{\"t[label]\": \"knows\", \"D[OUT]\":\"v[alice].id\", \"D[IN]\":\"v[bob].id\"}]"
+    And the traversal of
+      """
+      g.withStrategies(new PartitionStrategy(partitionKey: "_partition", writePartition: "a", readPartitions: ["a"])).
+        mergeE(xx1)
+      """
+    When iterated to list
+    Then the result should have a count of 1
+    And the graph should return 1 for count of "g.E().has(\"knows\",\"_partition\",\"a\")"
+    And the graph should return 1 for count of "g.E()"
+
+  @WithPartitionStrategy
+  Scenario: g_withStrategiesXPartitionStrategyXwrite_a_read_aXX_injectX0XmergeE
+    Given the empty graph
+    And the graph initializer of
+      """
+      g.addV("person").property("_partition","a").property("name","alice").
+        addV("person").property("_partition","a").property("name","bob")
+      """
+    And using the parameter xx1 defined as "m[{\"t[label]\": \"knows\", \"D[OUT]\":\"v[alice].id\", \"D[IN]\":\"v[bob].id\"}]"
+    And the traversal of
+      """
+      g.withStrategies(new PartitionStrategy(partitionKey: "_partition", writePartition: "a", readPartitions: ["a"])).
+        inject(0).
+        mergeE(xx1)
+      """
+    When iterated to list
+    Then the result should have a count of 1
+    And the graph should return 1 for count of "g.E().has(\"knows\",\"_partition\",\"a\")"
+    And the graph should return 1 for count of "g.E()"
+
+  @WithPartitionStrategy
+  Scenario: g_withStrategiesXPartitionStrategyXwrite_a_read_aXX_mergeVXlabel_person_name_aliceX_optionXonMatch_name_bobX
+    Given the empty graph
+    And the graph initializer of
+      """
+      g.addV("person").property("_partition","a").property("name","alice").
+        addV("person").property("_partition","b").property("name","alice")
+      """
+    And using the parameter xx1 defined as "m[{\"t[label]\": \"person\", \"name\":\"alice\"}]"
+    And using the parameter xx2 defined as "m[{\"name\":\"bob\"}]"
+    And the traversal of
+      """
+      g.withStrategies(new PartitionStrategy(partitionKey: "_partition", writePartition: "a", readPartitions: ["a"])).
```

Comment:
How `mergeV` will work when readPartitions not contains writePartition? 
something like
`g.withStrategies(new PartitionStrategy(partitionKey: "_partition", writePartition: "b", readPartitions: ["a"])).
        mergeV(xx1).option(Merge.onMatch, xx2)`

Reply:
it appears that if you do not specify that it will write a null for the partition. a `null` for graphs that don't support storage of it would be effectively ignored:

```text
gremlin> g = TinkerGraph.open().traversal()
==>graphtraversalsource[tinkergraph[vertices:0 edges:0], standard]
gremlin> g.addV().property('k',null)
==>v[0]
gremlin> g.V().elementMap()
==>[id:0,label:vertex]
```

so, the vertex would end up in no partition at all. for graphs that support null i guess it would write the partition as a `null` which i guess would then treat null as a partition value.

Reply:
oh - sorry - i misread what you wrote.  in the case you describe it's like a write only behavior. you can write stuff that you can't read. so for mergeV it would just always insert.